src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -73,6 +74,97 @@ get_pcidev(const xrt_core::device* device)
   return pcidev::get_dev(device->get_device_id(), device->is_userpf());
 }
 
+static query::sdm_sensor_info::result_type
+get_sdm_sensors(const xrt_core::device* device,
+                const query::sdr_req_type req_type,
+                const std::string path)
+{
+    auto pdev = get_pcidev(device);
+    query::sdm_sensor_info::result_type output;
+    query::sdm_sensor_info::data_type data;
+    std::vector<int> target_sysfs;
+    //All device sensors are stored in hwmon sysfs nodes with starts & ends as follows.
+    std::vector<std::string> start {"curr", "in", "power", "temp", "fan"};
+    std::vector<std::string> end {"label", "input", "max", "average", "highest"};
```

Comment:
You do not need so many dynamic allocations. Here everything is constant and local.
```suggestion
    std::array start {"curr", "in", "power", "temp", "fan"};
    std::array end {"label", "input", "max", "average", "highest"};
```
guessing we are up to C++17 now.